pkg/authenticator/dockersecret.go

```diff
@@ -0,0 +1,45 @@
+package authenticator
+
+import (
+	"fmt"
+	"os"
+)
+
+type dockersecret struct {
+}
+
+var _ Authenticator = (*dockersecret)(nil)
+
+func NewDockerSecret() *dockersecret {
+	return &dockersecret{}
+}
+
+func (s *dockersecret) GetAuthFile() (string, error) {
+	token, err := getSecretToken()
+	if err != nil {
+		return "", fmt.Errorf("Failed to get authfile %s\n", err)
+	}
+	authfile, err := createAuthFile(token)
+	if err != nil {
+		return "", fmt.Errorf("Failed to get authfile %s\n", err)
+	}
+
+	return authfile, nil
```

Comment:
Are you suggesting adding a volume mount which is used for just holding registry credentials?

Reply:
So I had this implemented in a branch here https://github.com/junshun/eks-anywhere-packages/tree/authn.

That was my original implementation, for it I had to change the NewHelm to take in mgr.config so then I can get the secret. The only reason I moved to this environment variable solution was it was easier to local testing and didn't require the rest.config to be passed into the helm driver. 

Currently my idea for specifing this enviroment variable is via the helm chart in the deployment.yaml so it would be a docker formated mounted secret but to the package controller pod it shows up as an environment variable
```
          env:
            - name: "DOCKER_CONFIG"
              valueFrom:
                secretKeyRef:
                  name: dockerconfig
                  key: .dockerconfigjson
```